pypit/arproc.py

```diff
@@ -604,62 +637,124 @@ def get_ampscale(slf, det, msflat):
 
 def get_datasec_trimmed(slf, fitsdict, det, scidx):
     """
-     Generate a frame that identifies each pixel to an amplifier, and then trim it to the data sections.
-     This frame can be used to later identify which trimmed pixels correspond to which amplifier
+    Primarily a wrapper with calls to get_datasec and pix_to_amp()
 
     Parameters
     ----------
-    slf : class
-      An instance of the ScienceExposure class
-    fitsdict : dict
-      Contains relevant information from fits header files
+    slf
+    fitsdict
     det : int
-      Detector number, starts at 1
     scidx : int
-      Index of science frame
 
     Returns
     -------
-    fitsdict : dict
-      Updates to the input fitsdict
+    Fills slf._datasect
+    fitsdict['naxis0'] and fittsdict['naxis1']
     """
     dnum = settings.get_dnum(det)
+    spectrograph = settings.argflag['run']['spectrograph']
+    scifile = fitsdict['directory'][scidx]+fitsdict['filename'][scidx]
+    numamplifiers = settings.spect[dnum]['numamplifiers']
+
+    # Instrument specific bits
+    if spectrograph in ['keck_lris_blue', 'keck_lris_red', 'keck_deimos']:
```

Comment:
A bit of an aside, but I think we want to minimize statements like this as much as possible.  As much as we can, we should handle instrument-specific operations like this at a higher level, instead of what looks like a fairly core, low-level function.  This may be easier said than done though, and this is where class objects can be useful...

Reply:
Agreed.  Will eliminate this in a future PR.

Added a TODO